File: ios/Classes/message/FileMessageNode.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK

class FileMessageNode: AbstractMessageNode {
    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let filePath: String = getParam(params: params, paramKey: "filePath")!;
        let fileName: String = getParam(params: params, paramKey: "fileName")!;
        return V2TIMManager.sharedInstance().createFileMessage(filePath, fileName: fileName)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[文件]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        let soundElem = elem as! V2TIMFileElem;
        let entity = FileMessageEntity();
        entity.uuid = soundElem.uuid;
        entity.fileName = soundElem.filename;
        entity.filePath = soundElem.path;
        entity.size = soundElem.fileSize;
        return entity;
    }
}

File: android/src/main/java/top/huic/tencent_im_plugin/message/FileMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMFileElem;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;

import top.huic.tencent_im_plugin.message.entity.FileMessageEntity;

/**
 * 文件消息节点
 */
public class FileMessageNode extends AbstractMessageNode<V2TIMFileElem, FileMessageEntity> {
    @Override
    public V2TIMMessage getV2TIMMessage(FileMessageEntity entity) {
        return V2TIMManager.getMessageManager().createFileMessage(entity.getFilePath(), entity.getFileName());
    }

    @Override
    public String getNote(V2TIMFileElem elem) {
        return "[文件]";
    }

    @Override
    public FileMessageEntity analysis(V2TIMFileElem elem) {
        return new FileMessageEntity(elem);
    }

    @Override
    public Class<FileMessageEntity> getEntityClass() {
        return FileMessageEntity.class;
    }
}
